src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -491,6 +491,9 @@ public IThreadPoolWorkItem Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool
 
         internal static bool Dispatch()
         {
+            // Put threadpool thread on default context
+            ExecutionContextSwitcher.SetDefaults();
```

Comment:
> So null means flow suppressed, unless you recapture when it means Default

I didn't follow that. I think `Capture` could stay as is, if it returns null it means EC != null and FlowSuppressed == true. If `Capture` returns `Default` it means EC == null. A current flow-suppressed context means EC != null.

> i.e. could have a FlowSupressed context with null being Default instead; but that's probably a more dangerous change?

I didn't follow that either, could you please elaborate?